### ISSUE TITLE
Put implicit parameter list in new line if characters in line exceeds the limit

### ISSUE THREAD

**Issue by jvican** [scala-ide/scalariform#196]
I think it would be very useful to have this configuration. Most of the times, when a method signature is very long, one wants to put in a new line a part of the declaration (and most of the times that's an implicit parameter list). Also, this new line is usually indented.

Example:

``` scala
def thisIsALongMethod[T](foo: String, bar: String, baz: Traversable[T])(implicit anotherArg: Foo[T]): Unit = { ??? }
```

would be converted into

``` scala
def thisIsALongMethod[T](foo: String, bar: String, baz: Traversable[T])
  (implicit anotherArg: Foo[T]): Unit = { ??? }
```

**Comment by fommil** [comment]
yes!!! So much yes

**Comment by fommil** [comment]
but maybe with aligned `(`

**Comment by jvican** [comment]
Both options could be accepted. What I don't like about the aligned `(` is that the start of the implicit parameter list will depend on the function name and function type signature, which most of the times ends up being huge because of type bounds. Therefore it would be like doing nothing as the parameter list will surpass the char limit anyway.

**Comment by jkinkead** [comment]
Line limits in general would be really, really nice. It's something I've halfheartedly looked into in the past.

**Comment by mjaghouri** [comment]
I want this feature too. Any news about whether anything like would be available soon?

**Comment by godenji** [comment]
@mjaghouri a fix for #77 is in progress. It won't be enforced automatically, but will allow param groups to exist on multiple lines (vs. forcing to single line as is the case now).

```
def foo[T]
  (a: A)
  (b: B)
  (implicit c: C)
```
This is in place on local dev branch. Working on Scala IDE integration, hope to have available within next couple of days, time permitting.

**Comment by godenji** [comment]
Fixed in 0.2.4, see the readme and [this comment](https://github.com/scala-ide/scalariform/issues/77#issuecomment-339398993) for details
